include/dlaf/matrix/tile.h

```diff
@@ -89,11 +163,16 @@ class Tile<const T, device> {
   /// @pre memory_view contains enough elements.
   /// The (i, j)-th element of the Tile is stored in the (i+ld*j)-th element of memory_view.
   Tile(const TileElementSize& size, memory::MemoryView<ElementType, device>&& memory_view,
-       SizeType ld) noexcept;
+       SizeType ld) noexcept
+      : data_(size, std::move(memory_view), ld) {}
+
+  Tile(TileDataType&& data) noexcept : data_(std::move(data)) {}
 
   Tile(const Tile&) = delete;
 
-  Tile(Tile&& rhs) noexcept;
+  Tile(Tile&& rhs) noexcept : data_(std::move(rhs.data_)), dep_tracker_(std::move(rhs.dep_tracker_)) {
+    rhs.dep_tracker_ = std::monostate();
+  };
```

Comment:
Could this and a few of the other constructors/operator= be defaulted instead? From a quick look it seems like they'd be equivalent. I think a moved-from variant will reset to the first variant (`monostate` in this case).

Reply:
No, the state of the variant remains the same, just the object inside is moved out.
https://godbolt.org/z/W74rjYb6M